### PR TITLE
Navigation bar items in methods

### LINKED ISSUE #6828: Cannot lookup symbol of nested function in method
_From @tinganho on January 31, 2016 17:22_

I cannot get a symbol of nested function in method.
```typescript
class A {
    b() {
       function test() {
           
       }
    }
}
```

I think the symbol `test` should show up when I do `CMD + SHIFT + O`. Since it works with nested functions:

```typescript
function test1 () {
    function test2() {
        
    }
}
```
I can get `test2` above.

_Copied from original issue: Microsoft/vscode#2584_

### PR DESCRIPTION
**Pull request by tinganho**
Fixes #6828

### CODE REVIEW

File: src/services/navigationBar.ts [hunk 1]
```diff
@@ -188,11 +197,19 @@ namespace ts.NavigationBar {
...
-                    // at module scope are 'top level'.
+                    // Or if it is not parented by another function(except for parent functions that
+                    // are methods and constructors). I.e all functions at module scope are 'top level'.
                     if (!isFunctionBlock(functionDeclaration.parent)) {
                         return true;
                     }
+                    else {
+                        const grandParentKind = functionDeclaration.parent.parent.kind;
+                        if (grandParentKind === SyntaxKind.MethodDeclaration ||
+                            grandParentKind === SyntaxKind.Constructor) {
```
Comment (DanielRosenwasser) [inline src/services/navigationBar.ts]: When won't `isFunctionBlock` cover these cases?

Comment (tinganho) [inline src/services/navigationBar.ts]: My assumption is never, it covers them and more function like nodes. That's why in the else clause I need to filter out only method declarations and constructors.

**Comment by tinganho** [comment]
Notice, I didn't made constructors and methods top level as a nested functions. So you cannot see where the parent declaration are.

Though I have an another implementation that does it in #7178. And I filter out method declarations and constructors, which does not have any function declarations inside it.

You can close this PR if you prefer #7178.

**Comment by tinganho** [comment]
Closing in favor of #7178 . Because that solution would work as if the method where nested functions.

**Pull request closed without merging**